src/top.rs

```diff
@@ -79,8 +110,18 @@ lazy_static! {
     static ref SENDER: Mutex<Option<Sender<GeckoMediaMsg>>> = {
         let (msg_sender, msg_receiver) = mpsc::channel();
         let (ok_sender, ok_receiver) = mpsc::channel();
+        let msg_sender_clone = msg_sender.clone();
         Builder::new().name("GeckoMedia".to_owned()).spawn(move || {
-            unsafe { GeckoMedia_Initialize(); }
+            let ptr = Box::into_raw(Box::new(msg_sender_clone));
```

Comment:
Can't the msg_sender_clone be passed directly instead of doing a dance with Box? :)

Reply:
I copied this pattern from [http://jakegoulding.com/rust-ffi-omnibus/objects/](http://jakegoulding.com/rust-ffi-omnibus/objects/). I think the Box is required to make this heap allocated.